src/org/aavso/tools/vstar/ui/MenuBar.java

```diff
@@ -1199,18 +1199,40 @@ public ActionListener createPluginManagerListener() {
 		return new ActionListener() {
 			@Override
 			public void actionPerformed(ActionEvent e) {
+				// Max: probably I'm paranoid: preventing opening PluginManagementDialog twice.
+				if (PluginManagementDialog.getPluginManagementActive()) return;
+				PluginManagementDialog.setPluginManagementActive(true);
```

Comment:
It's probably best to set this after line 1233 when the plugin manager has been fully activated. Indeed, one possible scenario is that the user clicks multiple times, causing this method to be invoked more than once, such that execution of a thread may be suspended after the active setter on line 1204 while a second enters the method, going no further than 1203 because the flag has already been set. The first thread would still resume to execute the rest and invoke the manager though, so probably okay. Adding the synchronised keyword to the method would allow the whole method to be treated as a critical section. That should work well with the active flag setting at line 1233. Am I making sense? It's later here. :)

Reply:
I think no. This code is intended to prevent the secondary execution of the menu action so it should be at the top of the "actionPerformed". The same check is done in PluginSettingsPane.java, the second place where one of the plugin management operations (deleteAllPlugins) is executed. "pluginManagementActive" flag can be placed elsewhere, PluginManagementDialog is not the unique place. pluginManagementActive flag is normally accessed from the main thread except one place: an exception handler at the line 1221 of MenuBar.jar where the flag is reset. This is why I declared it as "volatile" initially, yet synchronized getter and setter probably even better.